### PR TITLE
GetData now supports {start|end}Index/interval override

### PR DESCRIPTION
**Pull request by cfoisy-osisoft**
/preview endpoint which provided this function will be removed soon

### CODE REVIEW

File: ocs_samples/library_samples/Python3/ocs_sample_library_preview/Dataviews.py [hunk 1]
```diff
@@ -342,8 +355,12 @@ def getDataInterpolated(
         )
 
         next_page = response.headers.get("NextPage", None)
-        continuation_token = next_page[next_page.find("&continuationToken=")+19:] if next_page is not None else None
-        
+        continuation_token = (
+            next_page[next_page.find("&continuationToken=") + 19 :]
```
Comment (derekendres) [inline ocs_samples/library_samples/Python3/ocs_sample_library_preview/Dataviews.py]: 19?  is this interval -1

File: ocs_samples/library_samples/Python3/ocs_sample_library_preview/Dataviews.py [hunk 2]
```diff
@@ -247,6 +247,8 @@ def getDataviewPreview(
         value_class=None,
     ):
         """
+        --- Deprecated ---
```
Comment (derekendres) [inline ocs_samples/library_samples/Python3/ocs_sample_library_preview/Dataviews.py]: Any reason we should keep a deprecated function in a Sample for a product currently in preview?

File: ocs_samples/library_samples/Python3/ocs_sample_library_preview/Dataviews.py (hunk 1 above)
Comment (cfoisy-osisoft) [reply ocs_samples/library_samples/Python3/ocs_sample_library_preview/Dataviews.py]: This is to extract the continuationToken which is always 19 characters further than the index where this substring starts, plus is always the last piece of information returned in the next page URL.

File: ocs_samples/library_samples/Python3/ocs_sample_library_preview/Dataviews.py (hunk 2 above)
Comment (cfoisy-osisoft) [reply ocs_samples/library_samples/Python3/ocs_sample_library_preview/Dataviews.py]: Good question. This one is still in the currently published API ref, while the new parameters GetData do not show up yet. So I'll conservatively leave this in and remove it once both doc is updated and endpoint is removed.

File: ocs_samples/library_samples/Python3/setup.py [hunk 3]
```diff
@@ -15,8 +15,8 @@
     url="https://github.com/osisoft/OSI-Samples",
     packages=setuptools.find_packages(),    
     install_requires=[
-        'requests==2.21.0',
-        'python-dateutil==2.8.0'
+        'requests>=2.21.0',
```
Comment (derekendres) [inline ocs_samples/library_samples/Python3/setup.py]: Our policy with this is that we want to use the latest, but don't want to open it to unknown releases, which can include unwanted downstream dependencies or potential license changes

File: ocs_samples/library_samples/Python3/ocs_sample_library_preview/Dataviews.py (hunk 1 above)
Comment (cfoisy-osisoft) [reply ocs_samples/library_samples/Python3/ocs_sample_library_preview/Dataviews.py]: More robust code added

File: ocs_samples/library_samples/Python3/ocs_sample_library_preview/Dataviews.py (hunk 2 above)
Comment (cfoisy-osisoft) [reply ocs_samples/library_samples/Python3/ocs_sample_library_preview/Dataviews.py]: Removed

File: ocs_samples/library_samples/Python3/setup.py (hunk 3 above)
Comment (cfoisy-osisoft) [reply ocs_samples/library_samples/Python3/setup.py]: Pinned to latest 2.22.0

**Pull request merged**